org.adempiere.base/src/org/compiere/wf/MWFNextCondition.java

```diff
@@ -342,5 +364,40 @@ public MWFNextCondition markImmutable() {
 		makeImmutable();
 		return this;
 	}
+	
+	/**
+	 * Executes SQLStatement of Record and returns result as Object
+	 * @param Record_ID
+	 * @return SQL statement execution result
+	 * @throws Exception
+	 */
+	private Object getSqlResult(int Record_ID) throws Exception {
+		String sqlStatement = (String) get_Value("SQLStatement");
+		String tableName = getAD_Column().getAD_Table().getTableName();
+		String pkName = tableName + "_ID"; // primary key name

```

Comment:
not every table has primary key - so maybe would be better to check for this, also some tables have _ID different than table+_ID

Reply:
I've seen this construction many times in iDempiere core. I thought that it's a simple and relatively reliable method.

Reply:
There is a method to check that in MTable.isSingleKey() - and I think (not tested) you can obtain the key with MTable.getKeyColumns()[0]